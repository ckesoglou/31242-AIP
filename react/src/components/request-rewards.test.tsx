import React from "react";
import { shallow } from "enzyme";
import RequestRewards from "./request-rewards";

describe("<RequestRewards />", () => {
  it("should render correctly", () => {
    const wrapper = shallow(
      <RequestRewards
        requestID="1"
        items={[
          { id: "1", display_name: "Hug" },
          { id: "2", display_name: "Coffee" },
        ]}
        rewards={[
          { id: "1", display_name: "Hug" },
          { id: "2", display_name: "Coffee" },
          { id: "3", display_name: "Food" },
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
