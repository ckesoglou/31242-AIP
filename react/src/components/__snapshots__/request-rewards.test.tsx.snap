// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestRewards /> should render correctly 1`] = `
<div
  id="requestItem"
>
  <div
    id="multipleItemsContainer"
    onClick={[Function]}
  >
    <div
      className="firstCircle"
    >
      Hug
    </div>
    <div
      className="secondCircle"
    >
      Coffee
    </div>
    <div
      className="secondCircle thirdCircle"
    >
      +
    </div>
  </div>
  <WithStyles(ForwardRef(Popover))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    id="requestRewardPopover"
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <div
      id="requestRewardPopUp"
    >
      <div
        id="requestPopUpContent"
      >
        <WithStyles(ForwardRef(Typography))
          id="requestPopUpText"
        >
          Choose a reward to add
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Divider)) />
        <div
          id="requestAddContainer"
        >
          <RequestReward
            display_name="Hug"
            itemId="1"
            key="1"
            setSelectedReward={[Function]}
          />
          <RequestReward
            display_name="Coffee"
            itemId="2"
            key="2"
            setSelectedReward={[Function]}
          />
          <RequestReward
            display_name="Food"
            itemId="3"
            key="3"
            setSelectedReward={[Function]}
          />
        </div>
        <WithStyles(ForwardRef(Button))
          color="primary"
          id="addRewardButton"
          onClick={[Function]}
          variant="contained"
        >
          Add Reward?
        </WithStyles(ForwardRef(Button))>
      </div>
    </div>
  </WithStyles(ForwardRef(Popover))>
</div>
`;
